apps/ltn/src/route_planner.rs

```diff
@@ -126,22 +138,40 @@ impl RoutePlanner {
         panel.restore(ctx, &self.left_panel);
         self.left_panel = panel;
 
-        // Fade all neighbourhood interiors, so it's very clear when a route cuts through
-        let mut batch = GeomBatch::new();
-        for info in app.session.partitioning.all_neighbourhoods().values() {
-            batch.push(app.cs.fade_map_dark, info.block.polygon.clone());
-        }
+        let mut world = World::bounded(app.map.get_bounds());
+        self.waypoints.rebuild_world(ctx, &mut world, |x| x, 0);
+        world.initialize_hover(ctx);
+        world.rebuilt_during_drag(&self.world);
+        self.world = world;
+    }
+
+    // Called when waypoints changed, but the number has stayed the same. Aka, the common case of a
+    // waypoint being dragged. Does less work for speed.
+    fn update_minimal(&mut self, ctx: &mut EventCtx, app: &mut App) {
+        self.files.autosave(app);
+        let results_widget = self.recalculate_paths(ctx, app);
 
         let mut world = World::bounded(app.map.get_bounds());
-        world.draw_master_batch(ctx, batch);
         self.waypoints.rebuild_world(ctx, &mut world, |x| x, 0);
         world.initialize_hover(ctx);
         world.rebuilt_during_drag(&self.world);
         self.world = world;
+
+        self.left_panel.replace(ctx, "results", results_widget);
+
+        // TODO This is the most expensive part. While we're dragging, can we just fade out the
```

Comment:
If I take this part out, each update is about 0.01s. (aka it's about 0.02s just to do this) If the user isn't staring at the address that shows up in the card, this update is kind of useless anyway